src/main/java/org/hibernate/infra/asciidoctor/extensions/customnumbering/CustomNumberingProcessor.java

```diff
@@ -79,10 +79,19 @@ private void updateBlockCaption(StructuralNode block, String title, String secti
 
 		rubyObject.setInstanceVariable( "@caption", RubyString.newString(
 				Ruby.getGlobalRuntime(),
-				sectionNumber == NOT_NUMBERED_SECTION_NUMBER ?
-						String.format( "%s %d: ", title, indexNumber ) :
-						String.format( "%s %s.%d: ", title, sectionNumber, indexNumber )
+				String.format( "%s %s: ", title, getBlockNumber( sectionNumber, indexNumber ) )
```

Comment:
Let's put the block number in a variable instead of calling the method twice?

Reply:
force-pushed an update for this :)